src/main/java/com/sahokia/kos/jdbchomework/dao/TicketServiceDAO.java

```diff
@@ -0,0 +1,116 @@
+package com.sahokia.kos.jdbchomework.dao;
+
+import com.sahokia.kos.jdbchomework.connection.DatabaseConnection;
+import com.sahokia.kos.jdbchomework.model.Ticket;
+import com.sahokia.kos.jdbchomework.model.TicketType;
+import com.sahokia.kos.jdbchomework.model.User;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class TicketServiceDAO {
+
+    private final String SAVE_USER_QUERY = "INSERT INTO \"user\" (name, creation_date) VALUES (?, ?)";
+    private final String SAVE_TICKET_QUERY = "INSERT INTO \"ticket\" (user_id, ticket_type, creation_date) VALUES (?, ?::ticket_type, ?)";
+    private final String GET_USER_BY_ID_QUERY = "SELECT * FROM \"user\" WHERE id = ?";
+    private final String GET_TICKETS_BY_USER_ID_QUERY = "SELECT * FROM \"ticket\" WHERE user_id = ?";
+    private final String UPDATE_TICKET_TYPE_QUERY = "UPDATE \"ticket\" SET ticket_type = ?::ticket_type WHERE id = ?";
+    private final String DELETE_TICKETS_QUERY = "DELETE FROM \"ticket\" WHERE user_id = ?";
+    private final String DELETE_USER_QUERY = "DELETE FROM \"user\" WHERE id = ?";
+
+
+    public void saveUser(String name, Date creationDate) {
+        try (Connection connection = DatabaseConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SAVE_USER_QUERY)) {
+            preparedStatement.setString(1, name);
+            preparedStatement.setDate(2, creationDate);
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void saveTicket(int userId, TicketType ticketType, Date creationDate) {
+        try (Connection connection = DatabaseConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(SAVE_TICKET_QUERY)) {
+            preparedStatement.setInt(1, userId);
+            preparedStatement.setString(2, ticketType.toString());
+            preparedStatement.setDate(3, creationDate);
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public Optional<User> getUserById(int userId) {
+        try (Connection connection = DatabaseConnection.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_USER_BY_ID_QUERY)) {
+            preparedStatement.setInt(1, userId);
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    int id = resultSet.getInt("id");
+                    String name = resultSet.getString("name");
+                    Date creationDate = resultSet.getDate("creation_date");
+                    User user = new User(id, name, creationDate.toLocalDate());
+                    return Optional.of(user);
```

Comment:
Why do we need here Optional?)

Reply:
As I understand, using Optional is, in general, a good practice especially if the object does not exist in database. We want to return optional in this case to avoid null, but we also have to return optional even in cases when we find the object because of the method signture that we claim.